tests/integration/test_reactive.py

```diff
@@ -4,137 +4,227 @@
+    await reconcile(app_for_reactive, app_for_reactive.model)
+
+
+@pytest_asyncio.fixture(name="setup_queue", autouse=True)
 async def setup_queue_fixture(
     ops_test: OpsTest,
-    app_for_reactive: Application,
+    app: Application,
 ):
-    mongodb_uri = await _get_mongodb_uri(ops_test, app_for_reactive)
+    mongodb_uri = await _get_mongodb_uri(ops_test, app)
 
-    _clear_queue(mongodb_uri, app_for_reactive.name)
-    _assert_queue_is_empty(mongodb_uri, app_for_reactive.name)
+    _clear_queue(mongodb_uri, app.name)
+    _assert_queue_is_empty(mongodb_uri, app.name)
 
 
+@pytest.mark.abort_on_fail
 async def test_reactive_mode_spawns_runner(
     ops_test: OpsTest,
-    app_for_reactive: Application,
+    app: Application,
     github_repository: Repository,
     test_github_branch: Branch,
 ):
     """
-    arrange: A charm integrated with mongodb and a message is added to the queue.
+    arrange: Place a message in the queue and dispatch a workflow.
     act: Call reconcile.
-    assert: The message is consumed and a runner is spawned.
+    assert: A  runner is spawned to process the job and the message is removed from the queue.
     """
-    mongodb_uri = await _get_mongodb_uri(ops_test, app_for_reactive)
+    mongodb_uri = await _get_mongodb_uri(ops_test, app)
 
     run = await dispatch_workflow(
-        app=app_for_reactive,
+        app=app,
         branch=test_github_branch,
         github_repository=github_repository,
         conclusion="success",
         workflow_id_or_name=DISPATCH_TEST_WORKFLOW_FILENAME,
         wait=False,
     )
-    jobs = list(run.jobs())
-    assert len(jobs) == 1, "Expected 1 job to be created"
-    job = jobs[0]
-    job_url = job.url
-    job = JobDetails(
-        labels={app_for_reactive.name, "x64"},  # The architecture label should be ignored in the
-        # label validation in the reactive consumer.
-        url=job_url,
-    )
+    labels = {app.name, "x64"}  # The architecture label should be ignored in the
+    # label validation in the reactive consumer.
+    job = _create_job_details(run=run, labels=labels)
     _add_to_queue(
         json.dumps(json.loads(job.json()) | {"ignored_noise": "foobar"}),
         mongodb_uri,
-        app_for_reactive.name,
+        app.name,
     )
 
-    await reconcile(app_for_reactive, app_for_reactive.model)
+    # This reconcile call is to check that we are not killing machines that are under
+    # construction in a subsequent reconciliation.
+    await reconcile(app, app.model)
 
-    await wait_for_completion(run, conclusion="success")
-    _assert_queue_is_empty(mongodb_uri, app_for_reactive.name)
+    try:
+        await wait_for_completion(run, conclusion="success")
+    except TimeoutError:
+        assert False, (
+            "Job did not complete successfully, check the reactive log using tmate,"
+            " it might be due to infrastructure issues"
+        )
```

Comment:
This can be brittle if we run into infrastructure problems that cause a timeout when creating the runner. Therefore, I add a bit more explanatory message on failure.